integrated_channels/canvas/client.py

```diff
@@ -265,7 +276,114 @@ def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-
         # Todo: There isn't a great way for users to delete course completion data
         pass
 
+    def cleanup_duplicate_assignment_records(self, courses):
+        """
+        Placeholder
+        """
+        self._create_session()
+        failures = []
+        num_courses_removed = 0
+        for edx_course in courses:
+            canvas_course = CanvasUtil.find_course_by_course_id(
+                self.enterprise_configuration,
+                self.session,
+                edx_course
+            )
+
+            if not canvas_course:
+                failures.append(edx_course)
+                continue
+
+            canvas_assignments_url = CanvasAPIClient.course_assignments_endpoint(
+                canvas_base_url=self.enterprise_configuration.canvas_base_url,
+                course_id=canvas_course.get('id')
+            )
+            # Dict of most current, unique assignments in the course
+            current_assignments = {}
+
+            # Running list of duplicate assignments (ID's) that need to be deleted
+            assignments_to_delete = []
+
+            current_page_count = 0
+            more_pages_present = True
+            while more_pages_present and current_page_count < 150:
```

Comment:
I guess I would not want to delay this PR by asking in this work, but since we have the pagination follow pattern used in a couple places, and will use it more, will be some great refactoring to make it extracted out into a common place for canvas code to use ... a ticket for that?

Reply:
I've gone ahead and done it 😄